xds/src/main/java/io/grpc/xds/XdsRoutingLoadBalancer.java

```diff
@@ -190,8 +192,27 @@ protected Helper delegate() {
 
     @Override
     public PickResult pickSubchannel(PickSubchannelArgs args) {
-      // TODO(chengyuanzhang): to be implemented.
-      return PickResult.withError(Status.INTERNAL.withDescription("routing picker unimplemented"));
+      // Index ASCII headers by keys.
+      Map<String, Set<String>> asciiHeaders = new HashMap<>();
+      Metadata headers = args.getHeaders();
+      for (String headerName : headers.keys()) {
+        if (headerName.endsWith(Metadata.BINARY_HEADER_SUFFIX)) {
+          continue;
+        }
+        Set<String> headerValues = new HashSet<>();
+        Metadata.Key<String> key = Metadata.Key.of(headerName, Metadata.ASCII_STRING_MARSHALLER);
+        for (String value : headers.getAll(key)) {
+          headerValues.add(value);
+        }
+        asciiHeaders.put(headerName, headerValues);
+      }
+      for (Map.Entry<RouteMatch, SubchannelPicker> entry : routePickers.entrySet()) {
+        RouteMatch routeMatch = entry.getKey();
+        if (routeMatch.matches(args.getMethodDescriptor().getFullMethodName(), asciiHeaders)) {
```

Comment:
We should prepend '/' to `args.getMethodDescriptor().getFullMethodName()`, because the route's path or prefix in TD's config will always start with '/'. @menghanl Can you confirm?

Reply:
Path starts with "/", as defined in https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests

Reply:
> Path starts with "/", as defined in https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests

Yeah, java `MethodDescriptor.getFullMethodName()` does not include the leading '/', so we need prepend it.